src/Cli/dotnet/commands/dotnet-test/SolutionAndProjectUtility.cs

```diff
@@ -68,39 +84,68 @@ private static string[] GetSolutionFilePaths(string directory)
                 .ToArray();
         }
 
+        private static string[] GetSolutionFilterFilePaths(string directory)
+        {
+            return Directory.EnumerateFiles(directory, CliConstants.SolutionFilterExtensionPattern, SearchOption.TopDirectoryOnly).ToArray();
+        }
+
         private static string[] GetProjectFilePaths(string directory) => [.. Directory.EnumerateFiles(directory, CliConstants.ProjectExtensionPattern, SearchOption.TopDirectoryOnly).Where(IsProjectFile)];
 
         private static bool IsProjectFile(string filePath) => CliConstants.ProjectExtensions.Contains(Path.GetExtension(filePath), StringComparer.OrdinalIgnoreCase);
 
-        public static async Task<IEnumerable<string>> ParseSolution(string solutionFilePath, string directory)
+        public static string GetRootDirectory(string solutionOrProjectFilePath)
+        {
+            string fileDirectory = Path.GetDirectoryName(solutionOrProjectFilePath);
+            return string.IsNullOrEmpty(fileDirectory) ? Directory.GetCurrentDirectory() : fileDirectory;
+        }
+
+        public static IEnumerable<Module> GetProjectPropertiesInternal(string projectFilePath, ProjectCollection projectCollection)
+        {
+            var project = projectCollection.LoadProject(projectFilePath);
+            return ExtractModulesFromProject(project);
+        }
+
+        private static IEnumerable<Module> ExtractModulesFromProject(Project project)
```

Comment:
It's usually a good idea to have the return type as close as possible to the real type (with some exceptions for public apis)